R/look_up.R

```diff
@@ -0,0 +1,364 @@
+#' occdf <- look_up(occdf, int_key=interval_key, assign_with_GTS="GTS2012",
+#' early_interval = "stage", print_assigned=TRUE)
+#' }
+#' @export
+look_up <- function(occdf, int_key = palaeoverse::interval_key,
+                    assign_with_GTS = "GTS2020",
+                    early_interval = NULL, late_interval = NULL,
+                    print_assigned = FALSE) {
+
+
+
+  #=== Handling errors ===
+
+  if (is.data.frame(occdf) == FALSE) {
+    stop("`occdf` should be a dataframe.")
+  }
+
+  if (is.null(early_interval) && !("early_interval" %in% colnames(occdf))) {
+    stop("`occdf` needs to have a column named `early_interval`, or an
+         alternative name for the early interval column needs to be provided.")
+  }
+
+  if (!is.null(early_interval)) {
+    if (!is.character(early_interval))
+      stop("`early_interval` needs to be of type `character`")
+    if (!early_interval %in% colnames(occdf))
+      stop("`early_interval` needs to match a column name of `occdf`")
+  }
+
+  if (!is.null(late_interval)) {
+    if (!is.character(late_interval))
+      stop("`late_interval` needs to be of type `character`")
+    if (!late_interval %in% colnames(occdf))
+      stop("`late_interval` needs to match a column name of `occdf`")
+  }
+
+  if (is.data.frame(int_key) == FALSE) {
+    stop("`int_key` should be a dataframe.")
+  }
+
+  if (!("interval_name" %in% colnames(int_key))) {
+    stop("`int_key` needs to contain a column `interval_name`")
+  } else if (!is.character(int_key$interval_name)) {
+    stop("`int_key$interval_name` needs to be of type `character`")
+  }
+
+  if (!("early_stage" %in% colnames(int_key))) {
+    stop("`int_key` needs to contain a column `early_stage`")
+  } else if (!is.character(int_key$early_stage)) {
+    stop("`int_key$early_stage` needs to be of type `character`")
+  }
+
+  if ("late_stage" %in% colnames(int_key)) {
+    if (!is.character(int_key$late_stage)) {
+      stop("`int_key$late_stage` needs to be of type `character`")
+    }
+  }
+
+  if ("stage_max_ma" %in% colnames(int_key)) {
+    if (!is.numeric(int_key$stage_max_ma)) {
+      stop("`int_key$stage_max_ma` needs to be of type `numeric`")
+    }
+  }
+
+  if ("min_ma" %in% colnames(int_key)) {
+    if (!is.numeric(int_key$min_ma)) {
+      stop("`int_key$min_ma` needs to be of type `numeric`")
+    }
+  }
+
+  #=== Preparation ===
+
+  # save early and late int columns for easier handling
+  if (is.null(early_interval)) {
+    early <- occdf$early_interval
+  } else {
+    early <- occdf[, early_interval]
+  }
+
+  if (is.null(late_interval) && "late_interval" %in% colnames(occdf)) {
+    late <- occdf$late_interval
+  } else {
+    if (!is.null(late_interval)) {
+      late <- occdf[, late_interval]
+    } else { # if there is no `late` column, insert `early`
+      late <- early
+    }
+  }
+
+  # if there are missing values in `late`, fill them in from `early`
+  replace_pattern <- c("", " ")
```

Comment:
Good point. I have now added a warning if any NA, "", or " " are replaced from the late_interval column. This way the function can handle PBDB data, but warns the user in case these are genuine errors. But I'm also fine with removing it and just displaying the warning, then the users can sort this out themselves.